File: backend/YouTubeV2.Application/DTO/SearchDTOS/SearchResultsDto.cs
using YouTubeV2.Application.DTO.PlaylistDTOS;
using YouTubeV2.Application.DTO.UserDTOS;
using YouTubeV2.Application.DTO.VideoMetadataDTOS;

namespace YouTubeV2.Application.DTO.SearchDTOS
{
    public record SearchResultsDto(IReadOnlyList<VideoMetadataDto> videos,
                                   IReadOnlyList<UserDto> users,
                                   IReadOnlyList<PlaylistBaseDto> playlists);
}

File: frontend/src/app/core/models/search-results-dto.ts
import {UserDTO} from './user-dto'
import { VideoMetadataDto } from './video-metadata-dto'
import { PlaylistBaseDto } from './playlist-base-dto'

export interface SearchResultsDTO {
    videos: VideoMetadataDto [],
    users: UserDTO [],
    playlists: PlaylistBaseDto []
}